src/plugins/intel_gpu/src/kernel_selector/core/cl_kernels/reorg_yolo_gpu_ref.cl

```diff
@@ -41,25 +59,42 @@ KERNEL (reorg_yolo_ref)(const __global UNIT_TYPE* input, __global UNIT_TYPE* out
 
         output[dstIndex] = input[srcIndex];
     }
-#elif OUTPUT_LAYOUT_YXFB
-    int ic = get_global_id(0) / B;
-    int ib = get_global_id(0) % B;
-    int ih = get_global_id(2);
-    int iw = get_global_id(1);
+#else
+    const uint ic = get_global_id(2);
+    const uint ih = get_global_id(1);
+    const uint iw = get_global_id(0);
+
+    const uint OC = IC * STRIDE * STRIDE;
+    const uint OH = IH / STRIDE;
+    const uint OW = IW / STRIDE;
+
     for (int b = 0; b < B; b++) {
-        int dstIndex = ib + ic*B + ih*IC*B + iw*IH*IC*B;
+        const uint dstPlanarIndex = b*IC*IH*IW + ic*IH*IW + ih*IW + iw;
+        uint dstB, dstC, dstY, dstX;
+        FUNC_CALL(planar_to_bfyx)(dstPlanarIndex, B, OC, OH, OW, &dstB, &dstC, &dstY, &dstX);
+        const uint dstIndex = OUTPUT_GET_INDEX(dstB, dstC, dstY, dstX);
```

Comment:
Do we really need this `planar_to_bfyx` function? Won't something as follows work?

```cpp
        const int oc = ic % ic_off;
        const int offset = ic / ic_off;
        const int ow = iw * STRIDE + offset % STRIDE;
        const int oh = ih * STRIDE + offset / STRIDE;
        const uint dstIndex = OUTPUT_GET_INDEX(b, oc, oh, ow);
        const uint srcIndex = INPUT0_GET_INDEX(b, ic, ih, iw);
```
If it works, then I think both branches (`#if OUTPUT_LAYOUT_BFYX ... #else ...) can be unified

Reply:
Unfortunately, it doesn't work this way.  Reorg_yolo is a tricky operation - in the sense that it uses input indices to calculate output index and vice versa. It could be smth like 
```
        const uint dstIndex = OUTPUT_GET_INDEX(b, ic, ih, iw);
        const uint srcIndex = INPUT0_GET_INDEX(b, oc, oh, ow);
```
but this lead to out of bound indices.

Reply:
I see. May `${tensor_name}_GET_INDEX_SAFE` function help here? It applies modulo operation for each index before usage, like `(idx % actual_dimension_size) * dimension_offset`

Reply:
*_SAFE() macro don't work as well. 
While its return value is always valid for a given shape, at the same time it omits some values. 
For example, using b_fs_yx_fsv32 and input shape {1,4,2,2}, INPUT0_GET_INDEX_SAFE() return 0 for all the inputs (0, 0, 0, 0), (0, 0, 0, 1), (0, 0, 1, 0), (0, 0, 1, 1). As a result, some elements are never copied to output.